website/js/cloud.js

```diff
@@ -4,154 +4,153 @@
-                .fontSize(function(d) {
-                    return d.size;
-                })
-                .on("end", draw)
-                .start();
+        return mediaColor[key]
+      }
+      //Entering words
+    cloud.enter()
+      .append("text")
+      .style("font-family", "Impact")
+      // .style("fill", function(d, i) { return fill(i); })
+      .style("fill", function(d, i) {
+        return (d.size > 21 ? getColor(d) : 'lightgrey')
+      })
+      .style("font-weight", function(d) {
+        return (d.size > 16 ? 600 : 100)
+      })
+      .attr("text-anchor", "middle")
+      .attr('font-size', 1)
+      .attr('cursor', 'pointer')
+      .text(function(d) {
+        return d.text;
+      }).on("click", function(d) {
+        // createTimeline('#timeline', data)
+        // add word collection
+        var offset = $('#timeline').offset()
+        $('body,html').animate({ scrollTop: offset.top - 25 }, 'slow');
+
+        p1ClearCards()
+        news = report['words_count'][d.index][2]
+        dict = {}
+        for (i in mediaEN) {
+          dict[mediaEN[i]] = 0
+        }
+        for (var i in news) {
+          media = mediaC2EN[news[i]['media']]
+          dict[media]++
+            console.log(news[i]['title'] + " " + media)
+          if (dict[media] < 5)
+            p1AddNewsCard(media, news[i]['title'], "", news[i]['url'])
         }
+      });
+
+    //Entering and existing words
+    cloud
+      .transition()
+      .duration(600)
+      .style("font-size", function(d) {
+        return d.size + "px";
+      })
+      .attr("transform", function(d) {
+        return "translate(" + [d.x, d.y] + ")rotate(" + d.rotate + ")";
+      })
+      .style("fill-opacity", 1);
+
+    //Exiting words
+    cloud.exit()
+      .transition()
+      .duration(200)
+      .style('fill-opacity', 1e-6)
+      .attr('font-size', 1)
+      .remove();
+  }
+
+
+  //Use the module pattern to encapsulate the visualisation code. We'll
+  // expose only the parts that need to be public.
+  return {
+
+    //Recompute the word cloud for a new set of words. This method will
+    // asycnhronously call draw when the layout has been computed.
+    //The outside world will need to call this function, so make it part
+    // of the wordCloud return value.
+    update: function(words) {
+      d3.layout.cloud()
+        .size([width, 500])
+        .words(words)
+        .padding(2)
+        .rotate(function() {
+          return Math.random() * 90 - 45;
+        })
+        .font("Impact")
+        .fontSize(function(d) {
+          return d.size;
+        })
+        .on("end", draw)
+        .start();
     }
+  }
 }
 
 
 function showNewWords(vis, i) {
-    max = report['words_count'][0][1]
-    scale = max / 100
-    cloudConfig =  report['words_count'].map(function(obj,index) {
-        return { text: obj[0], size: (10 + obj[1] / scale),index:index};
-    });
-    vis.update(cloudConfig)
+  max = report['words_count'][0][1]
```

Comment:
If the key is constant, it better to use dot to get the value of object, e.g report.worlds_count[0][1].